datanode/api/trading_data_v2.go

```diff
@@ -3048,6 +3055,102 @@ func (t *tradingDataServiceV2) GetVegaTime(ctx context.Context, _ *v2.GetVegaTim
 
 // -- NetworkHistory --.
 
+func (t *tradingDataServiceV2) ExportNetworkHistory(req *v2.ExportNetworkHistoryRequest, stream v2.TradingDataService_ExportNetworkHistoryServer) error {
+	defer metrics.StartAPIRequestAndTimeGRPC("ExportNetworkHistory")()
+
+	if req.Table == v2.Table_TABLE_UNSPECIFIED {
+		return formatE(ErrNetworkHistoryNoTableName, errors.New("empty table name"))
+	}
+
+	tableName := strings.TrimPrefix(strings.ToLower(req.Table.String()), "table_")
+
+	segments, err := t.networkHistoryService.ListAllHistorySegments()
+	if err != nil {
+		return formatE(ErrListAllNetworkHistorySegment, err)
+	}
+
+	ch, err := networkhistory.GetContiguousHistory(segments, req.FromBlock, req.ToBlock)
+	if err != nil {
+		return formatE(ErrNetworkHistoryGetContiguousSegments, err)
+	}
+
+	header := metadata.Pairs("Content-Disposition", fmt.Sprintf("attachment;filename=vega-%s-%06d-%06d.csv", tableName, ch.HeightFrom, ch.HeightTo))
+	if err := stream.SendHeader(header); err != nil {
+		return formatE(ErrSendingGRPCHeader, err)
+	}
+
+	var chunk bytes.Buffer
+	var csvHeaders string
+	rowNum := 1
+
+	for _, segment := range ch.SegmentsOldestFirst {
+		segmentReader, err := t.networkHistoryService.GetHistorySegmentReader(stream.Context(), segment.GetHistorySegmentId())
+		if err != nil {
+			segmentReader.Close()
+			return formatE(ErrNetworkHistoryOpeningSegment, err)
+		}
+
+		segmentData, err := fsutil.ReadNetworkHistorySegmentData(segmentReader, tableName)
+		if err != nil {
+			segmentReader.Close()
+			return formatE(ErrNetworkHistoryExtractingSegment, err)
+		}
+
+		scanner := bufio.NewScanner(segmentData)
+		firstRowInSegment := true
+		for ; scanner.Scan(); rowNum++ {
+			row := scanner.Text()
+			if rowNum == 1 {
+				csvHeaders = row
+			}
+
+			// Don't repeat headers if they're the same in subsequent segments
+			if firstRowInSegment && rowNum != 1 && csvHeaders == row {
+				firstRowInSegment = false
+				continue
+			}
+
+			// If we're in a subsequent segment and headers have changed, write a blank line before the new headers
+			if firstRowInSegment && rowNum != 1 && csvHeaders != row {
```

Comment:
I made a thread on #project-datanode about this; it's kind of unpleasant because there's no way to do one HTTP request that triggers multiple downloads as far as I know. We'd have to make a couple of files and zip them up or something

Reply:
(the history segment reader I think should be able to return the schema version)

Reply:
Also most of the time when the overall schema version changes - the table being looked at won't have had any columns added or removed so it's just making life more complicated for the user.